korman/nodes/node_avatar.py

```diff
@@ -71,3 +71,388 @@ def export(self, exporter, bo, so):
+
+        for stage in self.find_inputs("stage_refs"):
+            animstage = plAnimStage()
+            animstage.animName = stage.anim_name
+            if stage.loop_option == "kLoopForever":
+                animstage.loops = -1
+            elif stage.loop_option == "kLoop":
+                animstage.loops = stage.num_loops
+
+            # Harvest additional AnimStage Settings, if available
+            settings = stage.find_input("stage_settings")
+            if settings:
+                animstage.forwardType = getattr(plAnimStage, settings.forward)
+                animstage.backType =getattr(plAnimStage, settings.backward)
+                animstage.advanceType = getattr(plAnimStage, settings.stage_advance)
+                animstage.regressType = getattr(plAnimStage, settings.stage_regress)
+                for flag in settings.notify_on:
+                    animstage.notify |= getattr(plAnimStage, flag)
+
+                advance_to = settings.find_input_socket("advance_to")
+                if advance_to.is_linked:
+                    # Auto-Advance to specific stage
+                    animstage.advanceTo = advance_to.links[0].from_node.stage_id
+                elif advance_to.auto_advance:
+                    # Auto-Advance
+                    animstage.advanceTo = None
+                else:
+                    # Don't Auto-Advance, just stop!
+                    animstage.advanceTo = -1
+
+                regress_to = settings.find_input_socket("regress_to")
+                if regress_to.is_linked:
+                    # Auto-Regress to specific stage
+                    animstage.regressTo = regress_to.links[0].from_node.stage_id
+                elif regress_to.auto_regress:
+                    # Auto-Regress
+                    animstage.regressTo = None
+                else:
+                    # Don't Auto-Regress, just stop!
+                    animstage.regressTo = -1
+
+            msbmod.addStage(animstage)
+
+    @property
+    def requires_actor(self):
+        return True
+
+
+class PlasmaAnimStageRefSocket(PlasmaAnimStageSocket, bpy.types.NodeSocket):
+    bl_color = (0.188, 0.186, 0.349, 1.0)
+
+    def draw(self, context, layout, node, text):
+        if isinstance(node, PlasmaMultiStageBehaviorNode):
+            try:
+                idx = next((idx for idx, socket in enumerate(node.find_input_sockets("stage_refs")) if socket == self))
+            except StopIteration:
+                layout.label(text)
+            else:
+                layout.label("Stage (ID: {})".format(idx))
+        else:
+            layout.label(text)
+
+
+class PlasmaSeekTargetSocket(PlasmaNodeSocketBase, bpy.types.NodeSocket):
+    bl_color = (0.180, 0.350, 0.180, 1.0)
+
+
+class PlasmaSeekTargetSocketIn(PlasmaSeekTargetSocket):
+    auto_target = BoolProperty()
+
+    def draw(self, context, layout, node, text):
+        if not self.is_linked:
+            layout.prop(self, "auto_target", text="Auto Smart Seek")
```

Comment:
It would be better, I think, to add the text and a description directly on the property.

Reply:
Oh, bizarre.  I must have just thrown this together (back during the previous run at this code) for testing.  Good catch, I'll put that back where it belongs.